File: src/leetcode/p1689/java/Solution.java
package leetcode.p1689.java;

public class Solution {
    public int minPartitions(String n) {
        int answer = 0;
        for (int i = 0; i < n.length(); i++) {
            if (answer < n.charAt(i) - '0') {
                answer = n.charAt(i) - '0';
            }
        }
        return answer;
    }
}

File: src/leetcode/p1689/kotlin/Solution.kt
package leetcode.p1689.kotlin

class Solution {
    fun minPartitions(n: String): Int {
        var answer = 0
        for (num in n) {
            if (answer < (num - '0')) {
                (num - '0').also { answer = it }
            }
        }
        return answer
    }
}